File: maxhanna.client/src/services/datacontracts/favourite/favourite.ts
export class Favourite { 
  id: number;
  url: string;
  imageUrl?: string;
  createdBy?: number;
  creationDate: Date;
  modifiedBy?: number;
  modificationDate: Date;
  lastAddedDate: Date;
  name?: string;
  userCount: number;
  accessCount: number;

  constructor(data: Partial<Favourite> = {}) { 
    this.id = data.id ?? 0;
    this.url = data.url ?? "";
    this.name = data.name ?? "";
    this.imageUrl = data.imageUrl ?? undefined;
    this.userCount = data.userCount ?? 0;
    this.createdBy = data.createdBy ?? undefined;
    this.creationDate = data.creationDate ? new Date(data.creationDate) : new Date();
    this.modifiedBy = data.modifiedBy ?? undefined;
    this.modificationDate = data.modificationDate ? new Date(data.modificationDate) : new Date();
    this.lastAddedDate = data.lastAddedDate ? new Date(data.lastAddedDate) : new Date();
    this.accessCount = data.accessCount ? data.accessCount : 0;
  }
}

File: maxhanna.Server/Controllers/DataContracts/Favourite/Favourite.cs
namespace maxhanna.Server.Controllers.DataContracts.Favourite
{
	public class Favourite
	{ 
		public int Id { get; set; }
		public string Url { get; set; }
		public string? Name { get; set; }
		public string? ImageUrl { get; set; }
		public int? UserCount { get; set; }
		public int? CreatedBy { get; set; }
		public DateTime CreationDate { get; set; }
		public int? ModifiedBy { get; set; }
		public DateTime ModificationDate { get; set; }
		public DateTime LastAddedDate { get; set; }
		public int? AccessCount { get; set; }

		public Favourite(int id, string url, string? name, string? imageUrl, int? userCount, 
		int? createdBy, DateTime creationDate, int? modifiedBy, DateTime modificationDate, DateTime lastAddedDate, int? accessCount)
		{ 
			Id = id;
			Url = url;
			Name = name;
			ImageUrl = imageUrl;
			UserCount = userCount;
			CreatedBy = createdBy;
			CreationDate = creationDate;
			ModifiedBy = modifiedBy;
			ModificationDate = modificationDate;
			LastAddedDate = lastAddedDate;
			AccessCount = accessCount;
		}
	}
}
